spec/workflow/payload_template_spec.rb

```diff
@@ -19,6 +19,24 @@
       it "returns the value from the inputs" do
         expect(subject.value(context, inputs)).to eq({"foo" => "bar", "bar" => "baz"})
       end
+
+      context "with an invalid payload" do
+        let(:payload) { {"foo.$" => "$.foo", "foo" => "bar"} }
+
+        it "raises an exception" do
+          expect { subject }.to raise_error(Floe::InvalidWorkflowError, "both foo.$ and foo present")
```

Comment:
nice

Reply:
Yeah I noticed this in the spec and realized it was a super easy validation to add now

Reply:
aah. here is a good example of where a separate validation step is probably better than a parse step. But still curious your take.